src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -82,6 +154,43 @@ public void SingleProcess_TryWaitMultipleTimesBeforeCompleting()
             Assert.True(p.WaitForExit(0));
         }
 
+        [Fact]
+        public async Task SingleProcess_TryWaitAsyncMultipleTimesBeforeCompleting()
```

Comment:
This isn't really doing what you want it do to, especially if you add the ThrowIfCancellationRequested I suggested earlier.  I think this shouldn't use a CancellationToken at all, e.g.
```C#
Process p = CreateProcessLong();
p.Start();

Task[] tasks = (from i in Enumerable.Range(0, 10) select p.WaitForExitAsync()).ToArray();
Assert.All(tasks, t => !t.IsCompleted);
p.Kill();
Task.WaitAll(tasks);
````

Reply:
I apologize, but I'm not following what it's doing versus should be doing. This test is basically an aync-ified clone of `SingleProcess_TryWaitMultipleTimesBeforeCompleting`.

Should the two tests diverge, or is there another test that should be added in addition?

Reply:
The token passed to WaitForExitAsync is already canceled prior to the call.  So it's going to get only to https://github.com/dotnet/runtime/pull/1278/files#diff-06c983bcaa17665febae824710b68dbeR1397 and never get any further.  If the goal is to test, for example, the event handle hookup and unhookup, this won't do that.  You would instead want to create a new CancellationTokenSource, pass its token to WaitForExitAsync, then cancel the token, and then await the previously returned task.

Reply:
Thanks for the additional info. I updated the unit test so that both cases are covered.

I also added comments to denote the different cases:
* Token is already canceled, so return synchronous
* Task is created, then token is canceled, so exercise cleanup code
* Task is awaited, then process is killed (the happy path)
* Waiting with a token on an already exited process, so return synchronous

Please let me know if there's anything else I missed. Thanks!